### ISSUE TITLE
Weird "Parse error: =" HLint error

### ISSUE THREAD

**Issue by rcook** [ndmitchell/hlint#236]
In a simple Haskell source file consisting of a `do` block and a single `let` binding I'm seeing an HLint "Parse error: =" error message:

```bash
$ cat Main.hs 
main = do
  let pattern = a
$ hlint Main.hs 
Main.hs:2:15: Error: Parse error: =
Found:
    main = do
      let pattern = a
  > 
  

1 hint
$ hlint --version
HLint v1.9.31, (C) Neil Mitchell 2006-2016
```

Interestingly, it looks like the binding has to be named `pattern` for this failure to occur. Clearly, there are many things wrong with this program, but I'm seeing this error on a real source file that GHC compiles without warnings or errors - this is the minimized reproducer of this issue. Is there something special about this identifier?

**Comment by ndmitchell** [comment]
Thanks for the small report @rcook, unfortunately it's a known issue, see #216.

Basically pass `-XNoPatternSynonyms` or avoid using the name "pattern" while we wait for a new haskell-src-exts release.
